src/databricks/labs/ucx/hive_metastore/data_objects.py

```diff
@@ -50,10 +54,33 @@ def _external_locations(self, tables: list[Row], mounts) -> list[ExternalLocatio
                         loc += 1
                     if not dupe:
                         external_locations.append(ExternalLocation(os.path.dirname(location) + "/"))
+                if location.startswith("jdbc"):
+                    pattern = r"(\w+)=(.*?)(?=\s*,|\s*\])"
+
+                    # Find all matches in the input string
+                    # Storage properties is of the format
+                    # "[personalAccessToken=*********(redacted), \
+                    #  httpPath=/sql/1.0/warehouses/65b52fb5bd86a7be, host=dbc-test1-aa11.cloud.databricks.com, \
+                    #  dbtable=samples.nyctaxi.trips]"
+                    matches = re.findall(pattern, table.storage_properties)
+
+                    # Create a dictionary from the matches
+                    result_dict = dict(matches)
+
+                    # Fetch the value of host from the newly created dict
+                    host = result_dict.get("host", "")
+                    logger.debug(f"Storage Properties is {table.storage_properties}")
+                    logger.debug(f"Host is {host}")
+                    jdbc_location = f"jdbc:databricks://{host}"
```

Comment:
Should not assume `databricks`
Also include the `httpPath`, `dbtable`, `query` keys but make sure to exclude credential type keys.